app/assets/javascripts/pageflow/editor/models/uploaded_file.js

```diff
@@ -10,6 +10,13 @@ pageflow.UploadedFile = Backbone.Model.extend({
 
     this.configuration.i18nKey = this.i18nKey;
 
+    if (pageflow.editor.nextUploadTargetFile){
+      this.set({parent_file_id: pageflow.editor.nextUploadTargetFile.get('id'),
+                parent_file_model_type: pageflow.editor.nextUploadTargetFile.fileType().typeName});
+      this.collection = pageflow.editor.nextUploadTargetFile
+        .nestedFiles(this.fileType().collectionName);
+    }
+
```

Comment:
This dependency on global editor state is ugly. This functionality should live inside of the `FilesCollection` which already knows the parent file and should set the collection property for new files all by itself.